tests/unit/game/effects/weakness.test.ts

```diff
@@ -0,0 +1,320 @@
+
+					expect(
+						game.components.find(
+							RowComponent,
+							query.row.currentPlayer,
+							query.row.index(0),
+						)?.health,
+					).toBe(VintageBeefCommon.health - ImpulseSVCommon.primary.damage)
+				},
+			},
+			{startWithAllCards: true, noItemRequirements: true},
+		)
+	})
+
+	test('Weakness Self Redundancy', () => {
+		testGame(
+			{
+				playerOneDeck: [VintageBeefCommon],
+				playerTwoDeck: [VintageBeefCommon, PotionOfWeakness, PotionOfWeakness],
+				saga: function* (game) {
+					yield* playCardFromHand(game, VintageBeefCommon, 'hermit', 0)
+					yield* endTurn(game)
+
+					yield* playCardFromHand(game, VintageBeefCommon, 'hermit', 0)
+					yield* playCardFromHand(game, PotionOfWeakness, 'single_use')
+					yield* endTurn(game)
+
+					yield* endTurn(game)
+
+					yield* playCardFromHand(game, PotionOfWeakness, 'single_use')
+					yield* attack(game, 'primary')
+					yield* endTurn(game)
+
+					expect(
+						game.components.find(
+							RowComponent,
+							query.row.currentPlayer,
+							query.row.index(0),
+						)?.health,
+					).toBe(
+						VintageBeefCommon.health - VintageBeefCommon.primary.damage - 20,
+					)
+				},
+			},
+			{startWithAllCards: true, noItemRequirements: true},
+		)
+	})
+
+	test('Weakness Heteroredundancy', () => {
+		testGame(
+			{
+				playerOneDeck: [SmallishbeansRare],
+				playerTwoDeck: [ImpulseSVCommon, PotionOfWeakness],
+				saga: function* (game) {
+					yield* playCardFromHand(game, SmallishbeansRare, 'hermit', 0)
+					yield* endTurn(game)
+
+					yield* playCardFromHand(game, ImpulseSVCommon, 'hermit', 0)
+					yield* playCardFromHand(game, PotionOfWeakness, 'single_use')
+					yield* attack(game, 'primary')
+					yield* endTurn(game)
+
+					expect(
+						game.components.find(
+							RowComponent,
+							query.row.currentPlayer,
+							query.row.index(0),
+						)?.health,
+					).toBe(SmallishbeansRare.health - ImpulseSVCommon.primary.damage - 20)
+				},
+			},
+			{startWithAllCards: true, noItemRequirements: true},
+		)
+	})
+
+	test('Weakness Lethality Check', () => {
+		testGame(
+			{
+				playerOneDeck: [ImpulseSVCommon, SmallishbeansRare, ImpulseSVCommon],
+				playerTwoDeck: [
+					SmallishbeansRare,
+					SmallishbeansRare,
+					SmallishbeansRare,
+					PotionOfWeakness
+				],
+				saga: function* (game) {
+					yield* playCardFromHand(game, ImpulseSVCommon, 'hermit', 0)
+					yield* playCardFromHand(game, SmallishbeansRare, 'hermit', 1)
+					yield* playCardFromHand(game, ImpulseSVCommon, 'hermit', 2)
+					yield* endTurn(game)
+
+					yield* playCardFromHand(game, SmallishbeansRare, 'hermit', 0)
+					yield* playCardFromHand(game, SmallishbeansRare, 'hermit', 1)
+					yield* playCardFromHand(game, SmallishbeansRare, 'hermit', 2)
+					yield* attack(game, 'secondary')
+					yield* endTurn(game)
+
+					yield* endTurn(game)
+
+					yield* playCardFromHand(game, PotionOfWeakness, 'single_use')
+					yield* attack(game, 'secondary')
```

Comment:
Potion of Weakness must be manually applied when played because it shows the confirmation modal.
```suggestion
					yield* playCardFromHand(game, PotionOfWeakness, 'single_use')
					yield* applyEffect(game)
					yield* attack(game, 'secondary')
```